tests/core/engine_adapter/test_duckdb.py

```diff
@@ -75,3 +75,19 @@ def test_set_current_catalog(make_mocked_engine_adapter: t.Callable, duck_conn):
     assert to_sql_calls(adapter) == [
         'USE "test_catalog"',
     ]
+
+
+def test_temporary_table(make_mocked_engine_adapter: t.Callable, duck_conn):
```

Comment:
Let's make sure to test this using a SQL example as well, besides calling the Python API methods.

Reply:
Also, we should test using a list as well, e.g. `"creatable_type": [exp.TemporaryProperty(), ...]`.

Reply:
Due to some complexity, we agreed to push this slightly later.